### ISSUE TITLE
Recommended hyperparameters for generating 30 second video

### ISSUE THREAD

**Issue by TheTinyTeddy** [SkyworkAI/SkyReels-V2#15]
Many thanks for the great work!

I was wondering if you could provide a inference bash script for generating long video (e.g. 30s) using the diffusion-forcing checkpoint, with recommended values for settings such as --num_frames --ar_step --overlap_history --addnoise_condition --causal_block_size etc. and their respective meanings in a bit more details if possible?

**Comment by linchunze** [comment]
model_id=Skywork/SkyReels-V2-DF-14B-540P
# synchronous inference
python3 generate_video_df.py \
  --model_id ${model_id} \
  --resolution 540P \
  --ar_step 0 \
  --base_num_frames 97 \
  --num_frames 257 \
  --overlap_history 17 \
  --prompt "A graceful white swan with a curved neck and delicate feathers swimming in a serene lake at dawn, its reflection perfectly mirrored in the still water as mist rises from the surface, with the swan occasionally dipping its head into the water to feed." \
  --addnoise_condition 20 \
  --offload

**Comment by TheTinyTeddy** [comment]
Thank you for the response!

BTW, what does the ar_step do if set to be >0?

**Comment by linchunze** [comment]
--num_frames: the total frame number you wanna to generate.
--overlap_history: the number of previous frames you wanna use to extend the video. 
--addnoise_condition: the noise level we add into clean overlap tokens in order to prevent error accumulation. 
--ar_step: if you set ar step > 0, the inference will be in asynchronous or "autoregressive" way, i.e. first tokens will be denoised ar_step times, then next tokens begin to be denoised. So ar_step > 0 may generate better results since next tokens can be guided by previous tokens.
--causal_block_size: used with ar_step>0, number of tokens you want to share the same denoising time step, e.g., causal_block_size==5 means you consider 5 tokens as block which will be denoised with the same time step.

**Comment by TheTinyTeddy** [comment]
Good to know! I have tried ar_step 2 and it did slightly improve the quality (especially object consistency) of the generated video. I have three more questions if you don't mind answering:
1) Why the frame size of 737 for generating 30 seconds video, it seems an odd number that is only divisible by 11, 67 which are also quite odd themselves.
2) Since asynchronous/autoregressive generation produces better quality, but why setting ar_step=0 by default if it could improve video quality? 
3) Does ar_step=0 imply no kv-cache usage, and ar_step>0 will start using kv-cache?

Many thanks for your time

**Comment by linchunze** [comment]
1. (737 = 1+ 736) frames will be encoded into (1 + 184 =185) tokens which is divisible by 5. As for asynchronous inference we recommend causal_block_size==5, the tokens number should be divisible by causal_block_size, that why we set 737 for 30s video.
2. It produces better results but it will be slightly slower. Asynchronous inference will take more steps to diffuse the whole sequence, so it will be slower than synchronous mode.
3. Our model is not a complete autoregressive model. The attention is still a bi-directional attention, so kv-cache is not applicable. You can further finetune the model with a causal mask attention then the model will become a complete autoregressive model and you can apply kv-cache for more efficient inference.

**Comment by XqZeppelinhead0702** [comment]
> Hi Thanks for your support! For generating longer videos, you can simply modify the --num_frames parameters. For example 30s you can set 737 frames.
> 
> model_id=Skywork/SkyReels-V2-DF-14B-540P
> 
> # synchronous inference
> python3 generate_video_df.py --model_id ${model_id} --resolution 540P --ar_step 0 --base_num_frames 97 --num_frames 737 --overlap_history 17 --prompt "A graceful white swan with a curved neck and delicate feathers swimming in a serene lake at dawn, its reflection perfectly mirrored in the still water as mist rises from the surface, with the swan occasionally dipping its head into the water to feed." --addnoise_condition 20 --offload

@linchunze Hello, could you please guide me how to set the hyperparameters for a 30-second videos when using the following inference script? Thanks!
```bash
# run Text-to-Video Generation
model_id=Skywork/SkyReels-V2-T2V-14B-540P
torchrun --nproc_per_node=2 generate_video.py \
  --model_id ${model_id} \
  --resolution 540P \
  --num_frames 97 \
  --guidance_scale 6.0 \
  --shift 8.0 \
  --fps 24 \
  --offload \
  --prompt "A serene lake surrounded by towering mountains, with a few swans gracefully gliding across the water and sunlight dancing on the surface." \
  --use_usp \
  --seed 42
```

**Comment by linchunze** [comment]
Generating a 540P video using the 14B model demands around 43.4GB peak VRAM.
